reconfigure: add arg check

gavanderhoorn opened the pull request:
Simple arg check for script.

Made a few too many typos (`puppet` does complain) and had to look up the docs a few too many times.

gavanderhoorn committed:
reconfigure: add usage check.

gavanderhoorn committed:
reconfigure: also check for valid role.

gavanderhoorn committed:
reconfigure: remove redundant arg empty check.

nuclearsandwich commented on reconfigure.bash:
```diff
@@ -2,6 +2,13 @@
 
 set -o errexit
 
+if [[ $# -ne 1 ]] || [[ $1 != "master" && $1 != "agent" && $1 != "repo" ]];
```
The test `[[ $# -ne 1 ]]` prevents the script from being run with no arguments when a role is already saved to the `role` file.

If possible I would like to preserve the ability to reconfigure without specifying an argument as it means we only need one common version of the reconfigure command line rather than one per role.

https://github.com/ros-infrastructure/buildfarm_deployment_config/blob/bf27c21108e74b1aa57ee4d6b9d8a6e7dc15fd05/hiera/hieradata/common.yaml#L28

nuclearsandwich committed:
Refactor usage check to allow no arguments variant.

nuclearsandwich committed:
Move variables above argument-check logic.

Only the script_dir is needed but moving all of them up keeps them
together and allows future checks to make use of the git repository and
branch info.

nuclearsandwich commented:
I pushed up some changes that should keep the checks you brought while still allowing the command to be run without an argument when a `role` file exists.

Merged this into my test deployment branch for testing.

nuclearsandwich committed:
Only check for role file when no argument is specified.

nuclearsandwich committed:
Fix typo in condition.

nuclearsandwich commented:
With apologies to @gavanderhoorn for littering their PR with iterative commits, this script as of 
2ab538d was used for a successful test deploy and subsequent reconfiguration of a host.
